src/PayU/PaymentsApi/PaymentsV4/Services/RequestBuilder.php

```diff
@@ -0,0 +1,291 @@
+            $clientData->setCommunicationLanguage($request->getUser()->getCommunicationLanguage());
+        }
+
+        $authorizationRequest = new AuthorizationRequest(
+            $request->getOrder()->getOrderRef(),
+            $request->getOrder()->getCurrency(),
+            $request->getOrder()->getBackRef(),
+            $authorizationData,
+            $clientData,
+            $this->getProductArray($request)
+        );
+
+        if ($request->getOrder()->getAirlineInfo() !== null) {
+            $flightSegments = $this->getFlightSegmentsArray($request);
+
+            $airlineInfo = new AirlineInfoData(
+                $request->getOrder()->getAirlineInfo()->getPassengerName(),
+                $flightSegments
+            );
+
+            $airlineInfo->setTicketNumber($request->getOrder()->getAirlineInfo()->getTicketNumber());
+            $airlineInfo->setRefundPolicy($request->getOrder()->getAirlineInfo()->getRestrictedRefund());
+            $airlineInfo->setReservationSystem($request->getOrder()->getAirlineInfo()->getReservationSystem());
+
+            if ($request->getOrder()->getAirlineInfo()->getTravelAgencyCode() !== null ||
+                $request->getOrder()->getAirlineInfo()->getTravelAgencyName() !== null
+            ) {
+                $travelAgency = new TravelAgency();
+                $travelAgency->setCode($request->getOrder()->getAirlineInfo()->getTravelAgencyCode());
+                $travelAgency->setName($request->getOrder()->getAirlineInfo()->getTravelAgencyName());
+
+                $airlineInfo->setTravelAgency($travelAgency);
+            }
+
+            $authorizationRequest->setAirlineInfoData($airlineInfo);
+        }
+
+        if ($request->getThreeDSecure() !== null) {
+            $mpi = new MpiData();
+
+            $mpi->setEci($request->getThreeDSecure()->getMpi()->getEci());
+            $mpi->setXid($request->getThreeDSecure()->getMpi()->getXid());
+            $mpi->setCavv($request->getThreeDSecure()->getMpi()->getCavv());
+            $mpi->setDsTransactionId($request->getThreeDSecure()->getMpi()->getDsTransactionId());
+            $mpi->setVersion($request->getThreeDSecure()->getMpi()->getVersion());
+
+            $threeDSecure = new ThreeDSecure($mpi);
+            $authorizationRequest->setThreeDSecure($threeDSecure);
+        }
+
+        if ($request->getStoredCredentials() !== null) {
+            $storedCredentials = new StoredCredentialsData();
+
+            $storedCredentials->setConsentType($request->getStoredCredentials()->getStoredCredentialsConsentType());
+            $storedCredentials->setUseType($request->getStoredCredentials()->getStoredCredentialsUseType());
+            $storedCredentials->setUseId($request->getStoredCredentials()->getStoredCredentialsUseId());
+
+            $authorizationRequest->setStoredCredentialsData($storedCredentials);
+        }
+
+        return json_encode($authorizationRequest);
```

Comment:
json_encode can also return false and we're not treating that case in any way. Maybe we could check before return if the result is false and throw RequestBuilderException, because false is not a valid return value

Reply:
done